Debug logs time

bukka opened the pull request:
#### Description

This PR adds a configurable time entry to the debug log. The first commit adds it as disable by default and the second one enables it by default (option can still be used to disable it in that case). I find it more useful to have time there as my use case involved correlation with other logs. But I understand if you prefer not to have it enabled by default. In such case I will remove the second commit.

<!-- Note: it is best to make pull requests from a branch rather than from main -->

#### Checklist

<!-- replace [ ] with [x] to select -->
<!-- (delete not applicable items) -->

- [x] Code modified for feature


#### Reviewer's checklist:

- [ ] Any issues marked for closing are addressed
- [x] There is a test suite reasonably covering new functionality or modifications
- [x] This feature/change has adequate documentation added
- [x] Code conform to coding style that today cannot yet be enforced via the check style test
- [x] Commits have short titles and sensible commit messages
- [ ] Coverity Scan has run if needed (code PR) and no new defects were found

simo5 commented:
@bukka Is there any good reason to make it optional ?

simo5 commented:
If we want to make it optional let's enable it if debug_level is higher, instead of adding another option to the already fragile debug string syntax, otherwise I would just simply always add timestamps.

bukka commented:
I don't think there is much point to make it optional. I added that just in case you prefer it without the time. I removed the option and did a quick check. Loggs look like:

```
[2025-03-10 17:28:19.676] [../src/provider.c:1494] OSSL_provider_init(): Provided config params:
....

simo5 commented:
It does make a log line much longer ... would it make sense to not only the time and not necessarily the date.
Or perhaps even frog the pretty print and log directly the unix-time integer?

I guess the question really is: do you need the actual date/time to correlate it with something else, or is relative difference between operations all that is needed here?

Not asing for further changes at this time, only trying to understand the goal.

bukka commented:
Yeah I need it mainly for correlation with other logs (nginx in my case). So using unix time integer would be a bit pain... But I can live with only keeping the time (although still prefer to keep milliseconds) if you prefer it but not sure if those 11 characters will make much difference. Some logs are already quite long anyway.

simo5 commented:
If you need correlation with external log, might as well have the whole date in it, and in a convenient format.
This is just debugging so probably better to have too much info than too little, people can easily parse the log file through sed and remove the dates if they are upset about the verbosity :)

simo5 commented:
Would you mind squashing the two commits?
Then I can merge.

bukka commented:
Thanks. Squashed.

simo5 commented:
thanks!
